mm.c

```diff
@@ -125,9 +133,40 @@ static size_t find_free_list_index(size_t words) {
 	physical memory with neigboring free blocks.
 	Returns the pointer to the beginning of this
 	new free block.
+	Coalesce is only called when a taken block is freed,
+	before the free block is placed into the free list.
+	As such, coalesce does not set pointer values.
 */
 static void *coalesce(void *bp) {
-
+	char **prev_block = PREV_BLOCK_IN_HEAP(bp);
+	char **next_block = NEXT_BLOCK_IN_HEAP(bp);
+	size_t prev_status = GET_STATUS(prev_block);
+	size_t next_status = GET_STATUS(next_block);
+	size_t new_size = GET_SIZE(bp);
+
+	if (prev_status == TAKEN && next_status == TAKEN) {
+		return bp;
+	} else if (prev_status == TAKEN && next_status == FREE) {
+		remove_block_from_free_list(next_block);
```

Comment:
As per the description in the comment above the function: "Coalesce is only called when a taken block is freed, before the free block is placed into the free list."

Thus, no.